plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -1071,4 +1127,97 @@ public <T> T calling(Callable<T> callable)
             this.statsGetter = requireNonNull(statsGetter, "statsGetter is null");
         }
     }
+
+    private class MetadataCacheAssertions
+    {
+        private final CachingJdbcClient cachingJdbcClient;
+        private final ConnectorSession session;
+        private final SchemaTableName firstName;
+        private final SchemaTableName secondName;
+
+        public MetadataCacheAssertions(CachingJdbcClient cachingJdbcClient, SchemaTableName firstName, SchemaTableName secondName)
+        {
+            this.cachingJdbcClient = requireNonNull(cachingJdbcClient, "cachingJdbcClient is null");
+            this.firstName = requireNonNull(firstName, "firstName is null");
+            this.secondName = requireNonNull(secondName, "secondName is null");
+            session = createSession("asession");
+        }
+
+        public void assertLoadsSchemaAndTableNames()
+        {
+            assertSchemaNamesCache(cachingJdbcClient).loads(1).misses(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getSchemaNames(session))
+                        .contains(firstName.getSchemaName())
+                        .doesNotContain(secondName.getSchemaName());
+            });
+            assertTableNamesCache(cachingJdbcClient).loads(1).misses(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getTableNames(session, Optional.empty()))
+                        .contains(firstName)
+                        .doesNotContain(secondName);
+            });
+            assertTableHandleByNameCache(cachingJdbcClient).misses(1).loads(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getTableHandle(session, firstName)).isNotEmpty();
+            });
+            assertTableHandleByNameCache(cachingJdbcClient).misses(1).loads(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getTableHandle(session, secondName)).isEmpty();
```

Comment:
If we are asserting only the schema and table names do we need this assertion related to `tableHandleByName`.

Reply:
It's needed to make sure that metadata caches have separate ttl from individual table caches.